src/zjs_modules.c

```diff
@@ -227,16 +230,23 @@ static ZJS_DECL_FUNC(native_print_handler)
     zjs_free(str);
     return ZJS_UNDEFINED;
 }
-
-static ZJS_DECL_FUNC(stop_js_handler)
+static void zjs_stop_js()
 {
-#ifdef CONFIG_BOARD_ARDUINO_101
-#ifdef CONFIG_IPM
-    zjs_ipm_free_callbacks();
-#endif
-#endif
     zjs_modules_cleanup();
+    zjs_remove_all_callbacks();
+    #ifdef CONFIG_BOARD_ARDUINO_101
+    #ifdef CONFIG_IPM
+        zjs_ipm_free_callbacks();
+    #endif
+    #endif
     jerry_cleanup();
+    jerry_init(JERRY_INIT_EMPTY);
+    zjs_modules_init();
+}
```

Comment:
I see that `javascript_stop()` in ashell/jerry-code.c duplicates some of this too. Could you refactor things to put that identical code in one place?

Reply:
Possibly, I'll look.

Reply:
Yeah was able to combine the two. Now the Ashell one calls the Modules one.  Only reason I still need a separate Ashell one is because it has a couple other things it needs to do that aren't related to the Modules one.